js/components/PauseControl.tsx

```diff
@@ -0,0 +1,49 @@
+import { h, Fragment } from 'preact';
+import { useCallback, useState } from 'preact/hooks';
+import { Apple2 as Apple2Impl } from 'js/apple2';
+import { ControlButton } from './ControlButton';
+
+/**
+ * PauseControl component properties.
+ */
+export interface PauseControlProps {
+    apple2: Apple2Impl | undefined;
+}
+
+/**
+ * Provides a control to pause and unpause the CPU.
+ *
+ * @param apple2 The Apple2 object
+ * @returns PauseControl component
+ */
+export const PauseControl = ({ apple2 }: PauseControlProps) => {
+    const [running, setRunning] = useState(true);
+
+    const doPause = useCallback(() => {
+        apple2?.stop();
+        setRunning(false);
+    }, [apple2]);
+
+    const doRun = useCallback(() => {
+        apple2?.run();
+        setRunning(true);
+    }, [apple2]);
```

Comment:
To me it seems like the pause/running control should be disabled while there is no `apple2` object as it has no effect and seems like it can get out of sync with the state in `apple2` otherwise. (In fact, I don't see anyway for a call to `apple2.run()` from some other code to change this pause/running control. Is that the case?)

Reply:
That would be polite, although the window during which there is no apple2 object is a couple of frames.

The ability for other code to pause/run and update the state of this widget introduces another layer of complexity I haven't yet fully tackled, global state, something that could be handled by something like Redux, but might also be addressed by a context.